app/enssubdomain/views.py

```diff
@@ -0,0 +1,257 @@
+    signer = Web3.toChecksumAddress(signer)
+    txn_hash = None
+    gasPrice = recommend_min_gas_price_to_confirm_in_time(1) * 10**9 if not settings.DEBUG else 15 * 10**9
+    subdomain = f"{github_handle}.{settings.ENS_TLD}"
+
+    transaction = {
+        'from': Web3.toChecksumAddress(settings.ENS_OWNER_ACCOUNT),
+        'value': 0,
+        'nonce': nonce,
+        'gas': 100000,
+        'gasPrice': gasPrice
+    }
+
+    resolver_contract = w3.eth.contract(
+        address=resolver_addr,
+        abi=resolver_abi,
+    )
+    txn = resolver_contract.functions.setAddr(
+        dot_eth_namehash(subdomain),
+        signer,
+        ).buildTransaction(transaction)
+    signed_txn = w3.eth.account.signTransaction(txn, private_key=settings.ENS_PRIVATE_KEY)
+    txn_hash = w3.eth.sendRawTransaction(signed_txn.rawTransaction) 
+
+    # hack to convert 
+    # "b'7bce7e4bcd2fea4d26f3d254bb8cf52b9ee8dd7353b19bfbc86803c27d9bbf39'"
+    # to "0x7bce7e4bcd2fea4d26f3d254bb8cf52b9ee8dd7353b19bfbc86803c27d9bbf39"
+    txn_hash = str(binascii.b2a_hex(txn_hash)).replace("b'","0x").replace("'","") 
+
+    return txn_hash
+
+
+def get_nonce():
+    web3_nonce = w3.eth.getTransactionCount(settings.ENS_OWNER_ACCOUNT)
+    next_db_nonce = 0
+    try:
+        last_ens = ENSSubdomainRegistration.objects.order_by('-end_nonce').first()
+        next_db_nonce = last_ens.end_nonce + 1
+    except:
+        pass
+
+    return max([web3_nonce, next_db_nonce])
+
+
+def handle_subdomain_post_request(request, github_handle):
+    # setup
+    signedMsg = request.POST.get('signedMsg', '')
+    signer = request.POST.get('signer', '').lower()
+    if signedMsg and signer:
+        # validation
+        message_hash = defunct_hash_message(text=f'Github Username : {github_handle}')
+        recovered_signer = w3.eth.account.recoverHash(message_hash, signature=signedMsg).lower()
+        if recovered_signer != signer:
+            return JsonResponse({'success': _('false'), 'msg': _('Sign Mismatch Error')})
+
+        # actually setup subdomain
+        start_nonce = get_nonce()
+        nonce = start_nonce
+        txn_hash_1 = set_owner(signer, github_handle, nonce)
+        nonce += 1
+        txn_hash_2 = set_resolver(signer, github_handle, nonce)
+        nonce += 1
+        txn_hash_3 = set_address_at_resolver(signer, github_handle, nonce)
+
+        profile = Profile.objects.filter(handle=github_handle).first()
+        ENSSubdomainRegistration.objects.create(
+            profile=profile,
+            subdomain_wallet_address=signer,
+            txn_hash_1=txn_hash_1,
+            txn_hash_2=txn_hash_2,
+            txn_hash_3=txn_hash_3,
+            pending=True,
+            signed_msg=signedMsg,
+            start_nonce=start_nonce,
+            end_nonce=nonce,
+            )
+        return JsonResponse(
+            {'success': _('false'), 'msg': _('Your request has been submitted. Please wait for the transaction to mine!')})
```

Comment:
@owocki The success needs to be `True` in this case and it need to be handled accordingly in the front-end. I can fix this if it can wait till Tuesday. thanks

Reply:
ill update now

Reply:
ericramos1980.gitcoin.eth